### PR TITLE
Add overlay-hover component

### PR DESCRIPTION
**Pull request by loudhearted**
This is an overlay for the hover/mouseover event; essentially a duplicate of OverlayClick but for hover.

### CODE REVIEW

File: src/js/components/overlays/overlay-click.es6 [hunk 1]
```diff
@@ -10,7 +10,8 @@ export default React.createClass({
     content: Type.node.isRequired,
     onClick: Type.func,
     onClose: Type.func,
-    onOpen: Type.func
+    onOpen: Type.func,
+    extraClasses: Type.string
```
Comment (HipsterBrown) [inline src/js/components/overlays/overlay-click.es6]: I thought we were using an Array of strings for the `extraClasses` props convention. That will work better with the `.concat` in the `.classes()` method later.

File: src/js/components/overlays/overlay-hover.es6 [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      </div>
+    );
+  }
+}
+
+OverlayHover.defaultProps = {
+  onHover: function() {},
+  onClose: function() {},
+  onOpen: function() {}
+}
```
Comment (HipsterBrown) [inline src/js/components/overlays/overlay-hover.es6]: If you stick with the `extraClasses` array convention, I would add an empty array as a default prop.

**Comment by HipsterBrown** [comment]
The `overlay-hover` event listeners should be a bit different than `overlay-click` because we don't want `mouseover` getting called on the whole document (probably a lot of events for that). See [this implementation in HCM](https://github.com/namely/namely/blob/9d8302efe9c0ca03cef5fde484370cf8010dd271/app/assets/javascripts/ui/time_off/summary/usage/available_now.es6#L67-L76) for how to show the content when the mouse is over the target element and hide it when `mouseout`.

**Comment by loudhearted** [comment]
@HipsterBrown Thanks for the heads up; I'll re-examine the implementation, and add array as one of the possible proptypes & as a default prop; since string does work, I'd like to keep it as an option since we do use both conventions.

**Pull request closed without merging**